scripts/build_source_deb.py

```diff
@@ -20,33 +20,45 @@
 import os
 import subprocess
 
-def get_buildpackage_version():
-    '''
-    Return the installed version of git-buildpackage as a tuple of three ints
-    '''
-    version_output = subprocess.check_output(['git-buildpackage', '--version'])
-    m = re.match('git-buildpackage\s+([0-9]+)\.([0-9]+)\.([0-9]+)\s*', version_output)
-    if m:
-        return tuple([int(x) for x in m.groups()])
-    return None
-
-# Parse command line args
-rosdistro, package, release_version = sys.argv[1:4]
-gbp_args = sys.argv[4:]
-
-# Create a git-buildpackage command appropriate for this slave
-version = get_buildpackage_version()
-print 'Detected git-buildpackage version' + str(version)
-if version[1] >= 6:
-    # This is a new version of git-buildpackage. It knows about the --git-upstream-tree arg
-    command = ['git-buildpackage', '-S', '--git-upstream-tree=TAG',
+def _get_package_subfolders(basepath, debian_package_name):
+    subfolders = []
+    for filename in os.listdir(basepath):
+        path = os.path.join(basepath, filename)
+        if not os.path.isdir(path):
+            continue
+        if filename.startswith('%s-' % debian_package_name):
+            subfolders.append(path)
+    return subfolders
+
+rosdistro, package, release_version, workdir = sys.argv[1:5]
+gbp_args = sys.argv[5:]
+
+try:
+	subfolders = _get_package_subfolders(workdir, debian_pkg)
+	assert len(subfolders) == 1, subfolders
+	sources_dir = subfolders[0]
+except:
+	sources_dir=workdir+'/build'
+
+cmd = ['gbp', 'buildpackage',
+	'--git-ignore-new',
+	'--git-ignore-branch',
+	# dpkg-buildpackage args
+	'-S']
+cmd += [
+	# dpkg-buildpackage args
+	'-us', '-uc']
+	# debuild args for lintian
+	#'--lintian-opts', '--suppress-tags', 'newer-standards-version']
+
+cmd += ['--git-upstream-tree=TAG',
         '--git-upstream-tag=release/{rosdistro}/{package}/{release_version}'.format(
             rosdistro=rosdistro, package=package, release_version=release_version)] + gbp_args
-else:
-    command = ['git-buildpackage', '-S'] + gbp_args
```

Comment:
it seems this PR drops support for older versions of git-buildpackage -- but I think that only affects 12.04 (based off a quick review of https://github.com/mikeferguson/buildbot-ros/pull/39 which introduced these files). Is this expected/tested to still work on 14.04?

Reply:
This configuration was successfully tested on a xenial machine and for kinetic repositories. Theoretically should be ok for trusty, I have a second buildbot system running on a 14.04 server, I hope I have time  this week to test this version and confirm that it works.

I showed you had before 2 command lines depending on the buildpackage version to support 12.04, alternatively we could have a get_linux_distro function and call "git-buildpackage" or "gpb buildpackage" depending on the value. Unfortunately I will not be able to test a 12.04 configuration.